packages/react/src/components/TreeView/TreeView-test.js

```diff
@@ -1,95 +1,219 @@
-      expect(onTreeSelect).toHaveBeenCalledTimes(1);
+    it('should respect selected prop', () => {
+      render(
+        <TreeView label="Tree View" selected={['Node 1']}>
+          <TreeNode id="Node 1" data-testid="Node 1" label="Node 1" />
+        </TreeView>
+      );
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(screen.getByRole('treeitem', { name: 'Node 1' })).toHaveClass(
+        'cds--tree-node--selected'
+      );
     });
 
-    it('should handle selection at the node level', () => {
-      wrapper.find('TreeNode[value="2"]').simulate('click');
-      expect(onTreeSelect).toHaveBeenCalledTimes(1);
-      expect(onNodeSelect).toHaveBeenCalledTimes(1);
+    it('should respect size prop', () => {
+      render(<TreeView size="xs" label="Tree" />);
+
+      const tree = screen.getByRole('tree');
+
+      expect(tree).toHaveClass('cds--tree--xs');
     });
   });
 
-  describe('Tree node expansion', () => {
-    it('Caret icon should not render in leaf nodes', () => {
-      expect(wrapper.find(CaretDown).length).toBe(2);
+  describe('behaves as expected', () => {
+    it('should render tree with two nodes', () => {
+      render(
+        <TreeView label="Tree View">
+          <TreeNode label="Node 1" />
+          <TreeNode label="Node 2" />
+        </TreeView>
+      );
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(screen.getByText('Node 1')).toBeInTheDocument();
+      expect(screen.getByText('Node 2')).toBeInTheDocument();
+    });
+
+    it('should render tree with expanded node', () => {
+      render(
+        <TreeView label="Tree View">
+          <TreeNode data-testid="Node 1" label="Node 1" isExpanded={true}>
+            <TreeNode data-testid="Node 2" label="Node 2" />
+          </TreeNode>
+        </TreeView>
+      );
+
+      const nodeParent = screen.getByTestId('Node 1');
+      const nodeChild = nodeParent?.querySelector('div > span');
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(nodeChild).toHaveClass('cds--tree-parent-node__toggle');
+      expect(within(nodeParent).getByText('Node 1')).toBeInTheDocument();
+      expect(within(nodeParent).getByText('Node 2')).toBeInTheDocument();
+    });
+
+    it('should render tree with disabled nodes', () => {
+      render(
+        <TreeView label="Tree View">
+          <TreeNode label="Node 1" data-testid="Node 1" />
+          <TreeNode
+            data-testid="Node 2"
+            label="Node 2"
+            disabled={true}
+            isExpanded={true}>
+            <TreeNode isExpanded={true} data-testid="Node 3" label="Node 3" />
+          </TreeNode>
+        </TreeView>
+      );
+
+      const nodeParent = screen.getByTestId('Node 1');
+      const nodeChild = screen.getByTestId('Node 2');
+
+      expect(screen.getByLabelText('Tree View')).toBeInTheDocument();
+      expect(nodeChild).toHaveClass('cds--tree-node--disabled');
```

Comment:
Could this instead query against the actual disabled attribute? Or aria-disabled? I'm not sure which the component uses.